DS_and_Algorithm/BST.py

```diff
@@ -0,0 +1,33 @@
+class TreeNode:
+    def __init__(self, key):
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/Amruta101998/DS-and-Algorithm/pull/11#issuecomment-2262885626>#6831b4</a><span> - 08/01/2024, 12:16 pm</span>
<br>

<div><small>🟠</small> Medium importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The '__init__' method lacks type hints, which can improve code readability and maintainability.
</div>
<div id="fix">
<b>Fix:</b> Add type hints to the '__init__' method to specify the expected type of the 'key' parameter.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -1,5 +1,5 @@
  class TreeNode:
 -    def __init__(self, key):
 +    def __init__(self, key: int):
         self.left = None
         self.right = None
         self.val = key
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged